landlab/components/bedrock_landslider/bedrock_landslider.py

```diff
@@ -0,0 +1,669 @@
+            a list as a property of a bedrock_landslider object.
+        store_landslides_volume : bool , optional
+            Store the volume of landslides through time. Values will be stored in
+            a list as a property of a bedrock_landslider object.
+        store_landslides_volume_sed  : bool , optional
+            Store the volume of eroded bedrock through time. Values will be
+            stored in a list as a property of a bedrock_landslider object.
+        store_landslides_volume_bed : bool  , optional
+            Store the volume of eroded sediment landslides through time.
+            Values will be stored in a list as a property of a
+            bedrock_landslider object.
+        """
+        super(BedrockLandslider, self).__init__(grid)
+
+        topo = self.grid.at_node["topographic__elevation"]
+        soil = self.grid.at_node["soil__depth"]
+
+        if "bedrock__elevation" not in grid.at_node:
+            grid.add_full("bedrock__elevation", topo - soil, at="node", dtype=float)
+
+        # Check consistency of bedrock, soil and topographic elevation fields
+        if not np.allclose(
+            grid.at_node["bedrock__elevation"] + grid.at_node["soil__depth"],
+            grid.at_node["topographic__elevation"],
+        ):
+            raise RuntimeError(
+                "The sum of bedrock elevation and topographic elevation should be equal"
+            )
+
+        self.initialize_output_fields()
+
+        # Store grid and parameters
+        self._angle_int_frict = angle_int_frict
+        self._cohesion_eff = cohesion_eff
+        self._rho_r = rho_r
+        self._grav = grav
+        self._fraction_fines_LS = fraction_fines_LS
+        self._phi = phi
+        self._landslides_return_time = landslides_return_time
+        self._max_pixelsize_landslide = max_pixelsize_landslide
+        self._verbose_landslides = verbose_landslides
+        self._landslides_on_boundary_nodes = landslides_on_boundary_nodes
+        self._critical_sliding_nodes = critical_sliding_nodes
+
+        # Data structures to store properties of simulated landslides.
+        self._landslides_size = [] if store_landslides_size else None
+        self._landslides_volume = [] if store_landslides_volume else None
+        self._landslides_volume_sed = [] if store_landslides_volume_sed else None
+        self._landslides_volume_bed = [] if store_landslides_volume_bed else None
+
+        # Check input values
+        if phi >= 1.0 or phi < 0.0:
+            raise ValueError(f"Porosity must be between 0 and 1 ({phi})")
+
+        if fraction_fines_LS > 1.0 or fraction_fines_LS < 0.0:
+            raise ValueError(
+                f"Fraction of fines must be between 0 and 1 ({fraction_fines_LS})"
+            )
+
+        # Set seed
+        if seed is not None:
+            np.random.seed(seed)
+
+    # Getters for properties
+    @property
+    def landslides_size(self):
```

Comment:
Add a docstring with just a one line description for each of these properties.

Reply:
Done